bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/target/obsComp/PwfsGuideProbe.java

```diff
@@ -284,41 +284,47 @@ public GuideStarValidation validate(Coordinates coords, ObsContext ctx) {
         return GuideProbeUtil.instance.validate(coords, this, ctx);
     }
 
-    public Option<BoundaryPosition> checkBoundaries(SPTarget guideStar, ObsContext ctx){
+    public Option<BoundaryCheck> checkBoundaries(SPTarget guideStar, ObsContext ctx){
         return guideStar
             .getSkycalcCoordinates(ctx.getSchedulingBlockStart())
             .flatMap(cs -> checkBoundaries(cs, ctx));
     }
 
     /**
-     * Returns BoundaryPosition.outside, if the coordinates are outside the outer limit,
-     * inside, if inside the inner limit and innerBoundary if between the inner and outer limits
-     * (which is where the PWFS should normally be).
+     * Returns:
+     *
+     * - BoundaryCheck.outOfRange if outside the outer limit.
+     *
+     * - BoundaryCheck.nearBoundary if in between the inner limit and the inner boundary.
+     *
+     * - BoundaryCheck.vignetting if it obscures the science area.
+     *
+     * - BoundaryCheck.inRange if inside the inside limit.
      *
      * @param coords the coordinates
      * @param ctx the context
      */
-    public Option<BoundaryPosition> checkBoundaries(final Coordinates coords, final ObsContext ctx) {
+    public Option<BoundaryCheck> checkBoundaries(final Coordinates coords, final ObsContext ctx) {
         return ctx.getBaseCoordinates().map(baseCoordinates -> {
             final Angle positionAngle = ctx.getPositionAngleJava();
             final Set<Offset> sciencePositions = ctx.getSciencePositions();
 
             // check positions against corrected outer patrol field bounds
             return getCorrectedPatrolField(ctx).map(pf -> {
                 final BoundaryPosition bp = pf.checkBoundaries(coords, baseCoordinates, positionAngle, sciencePositions);
-                if (bp != BoundaryPosition.inside) {
-                    return BoundaryPosition.outside;
+                if ((bp != BoundaryPosition.inside) && (bp != BoundaryPosition.innerBoundary)) {
+                    return BoundaryCheck.nearBoundary;
```

Comment:
This should continue to be `outside`, I believe.  If it isn't one of the inside positions, it's outside.  If we need to distinguish `innerBoundary` though we could switch on `bp` here I guess.  I can't remember the syntax but something like:

```java
switch (bp) {
    case BoundaryPosition.outside:
    case BoundaryPosition.outerBoundary:
        return outOfRange;
    case BoundaryPosition.innerBoundary:
        return nearBoundary;
    case BoundaryPosition.inner:
       // here we have to check whether we're vignetting to decide between inRange and vignetting
      ....
}

```